src/test/java/edu/hm/hafner/analysis/IssueFilterTest.java

```diff
@@ -0,0 +1,330 @@
+    private static final String REGEX_MATCH = ".*#";
+
+    private static final Issue ISSUE_1 = new IssueBuilder()
+            .setFileName("filter_me_by_file#1")
+            .setPackageName("filter_me_by_package#1")
+            .setModuleName("filter_me_by_module#1")
+            .setCategory("filter_me_by_category#1")
+            .setType("filter_me_by_type#1")
+            .build();
+
+    private static final Issue ISSUE_2 = new IssueBuilder()
+            .setFileName("filter_me_by_file#2")
+            .setPackageName("filter_me_by_package#2")
+            .setModuleName("filter_me_by_module#2")
+            .setCategory("filter_me_by_category#2")
+            .setType("filter_me_by_type#2")
+            .build();
+
+    private static final Issue ISSUE_3 = new IssueBuilder()
+            .setFileName("filter_me_by_file#3")
+            .setPackageName("filter_me_by_package#3")
+            .setModuleName("filter_me_by_module#3")
+            .setCategory("filter_me_by_category#3")
+            .setType("filter_me_by_type#3")
+            .build();
+
+    private static final Issue ISSUE_4 = new IssueBuilder()
+            .setFileName("filter_me_by_file#4")
+            .setPackageName("filter_me_by_package#4")
+            .setModuleName("filter_me_by_module#4")
+            .setCategory("filter_me_by_category#4")
+            .setType("filter_me_by_type#4")
+            .build();
+
+    private static final Issue ISSUE_5 = new IssueBuilder()
+            .setFileName("filter_me_by_file#5")
+            .setPackageName("filter_me_by_package#5")
+            .setModuleName("filter_me_by_module#5")
+            .setCategory("filter_me_by_category#5")
+            .setType("filter_me_by_type#5")
+            .build();
+
+    private static final Issues ISSUES = new Issues(asList(ISSUE_1, ISSUE_2, ISSUE_3, ISSUE_4, ISSUE_5));
+
+    @Test
+    void emptyListWithEmptyFiltersShouldRemainEmpty() {
+        final IssueFilter issueFilter = new IssueFilter();
+
+        assertThat(issueFilter.applyFilters(new Issues()))
+                .as("Empty list should remain empty after applying no filters.")
+                .hasSize(0);
+    }
+
+    @Test
+    void emptyListWhenFiltersSetShouldRemainEmpty() {
+        final IssueFilter issueFilter =
+                new IssueFilter()
+                        .addIncludedFiles(match(1, 3))
+                        .addExcludedFiles(match(1, 2));
+
+        assertThat(issueFilter.applyFilters(new Issues()))
+                .as("Empty list should remain empty after applying filters.")
+                .hasSize(0);
+    }
+
+    @Test
+    void issuesShouldRemainEqualWhenNoFiltersSet() {
+        final IssueFilter issueFilter = new IssueFilter();
+
+        assertThat(issueFilter.applyFilters(ISSUES))
+                .as("No issues should be filtered out when no filters are set.")
+                .hasSize(ISSUES.getSize());
+    }
+
+    @Test
+    void noIssuesShouldBeFilteredWhenNoFiltersMatch() {
```

Comment:
Dieser Test führt semantisch in die Irre: es wird ja ein String Match gebildet und keine Zahlen verglichen...